shared/templates/create_permissions.py

```diff
@@ -0,0 +1,224 @@
+#
+# create_permission.py
+#        generate template-based checks for file permissions/ownership
+
+import re
+
+from template_common import FilesGenerator, UnknownTargetError
+
+
+class PermissionGenerator(FilesGenerator):
+    def generate(self, target, path_info):
+        # the csv file contains lines that match the following layout:
+        #    directory,file_name,uid,gid,mode[,preferred_id]
+
+        dir_path, file_name, uid, gid, mode = path_info[0:5]
+
+        if uid == "" and gid == "" and mode == "":
+            # If UID, GID, and mode are not defined, skip processing this file
+            return
+
+        # build a string out of the path that is suitable for use in id tags
+        # example:	/etc/resolv.conf --> _etc_resolv_conf
+        # path_id maps to FILEID in the template
+
+        if len(path_info) == 6:
+            path_id = "_" + path_info[5]
```

Comment:
@yuumasato Since all of the other IDs are path-based, they begin with a starting `/`; hence they'll be converted to a leading `_`, making the IDs look good and function as expected.

In user-supplied IDs, we didn't prefix them with an underscore, so now we do. :) Good catch, thanks!